Documentation/ColumnsConfig/Type/User.rst

```diff
@@ -45,51 +45,142 @@ The `user` field can be useful, if:
-        {
-            // Custom TCA properties and other data can be found in $this->data, for example the above
-            // parameters are available in $this->data['parameterArray']['fieldConf']['config']['parameters']
-            $result = $this->initializeResultArray();
-            $result['html'] = 'my map content';
-            return $result;
-        }
-    }
+   Add the field to the TCA definition, here in
+   :file:`Configuration/TCA/Overrides/fe_users.php`::
 
+      'tx_examples_special' => [
+         'exclude' => 0,
+         'label' => 'LLL:EXT:examples/Resources/Private/Language/locallang_db.xlf:fe_users.tx_examples_special',
+         'config' => [
+            'type' => 'user',
+            // renderType needs to be registered in ext_localconf.php
+            'renderType' => 'specialField',
+            'parameters' => [
+               'size' => '30',
+               'color' => '#F49700',
+            ],
+         ],
+      ],
+
+3. Implement the FormElement class
+
+   The :php:`renderType` can be implemented by extending the class
+   :php:`AbstractFormElement` and overriding the function :php:`render()`.
+
+      <?php
+      declare(strict_types = 1);
+      namespace T3docs\Examples\Form\Element;
+
+      use TYPO3\CMS\Backend\Form\Element\AbstractFormElement;
+      use TYPO3\CMS\Core\Utility\GeneralUtility;
+      use TYPO3\CMS\Core\Utility\StringUtility;
+
+      class SpecialFieldElement extends AbstractFormElement
+      {
+         public function render():array
+         {
+            $row = $this->data['databaseRow'];
+            $parameterArray = $this->data['parameterArray'];
+            $color = $parameterArray['fieldConf']['config']['parameters']['color'];
+            $size = $parameterArray['fieldConf']['config']['parameters']['size'];
+
+            $fieldInformationResult = $this->renderFieldInformation();
+            $fieldInformationHtml = $fieldInformationResult['html'];
+            $resultArray = $this->mergeChildReturnIntoExistingResult($this->initializeResultArray(), $fieldInformationResult, false);
+
+            $fieldId = StringUtility::getUniqueId('formengine-textarea-');
+
+            $attributes = [
+               'id' => $fieldId,
+               'name' => htmlspecialchars($parameterArray['itemFormElName']),
+               'size' => $size,
+               'data-formengine-input-name' => htmlspecialchars($parameterArray['itemFormElName']),
+               'onChange' => implode('', $parameterArray['fieldChangeFunc']),
+            ];
+
+            $attributes['placeholder'] = 'Enter special value for user "'.htmlspecialchars(trim($row['username'])).
+               '" in size '. $size;
+            $classes = [
+               'form-control',
+               't3js-formengine-textarea',
+               'formengine-textarea',
+            ];
+            $itemValue = $parameterArray['itemFormElValue'];
+            $attributes['class'] = implode(' ', $classes);
```

Comment:
Is there a specific reason why not all of this is done before / during `$attributes` initialization?

Reply:
@linawolf Can you have a look at this?

Reply:
The code is taken from the examples extension and is proven as working. You can make a PR there if you want to change it.